src/sagemaker_xgboost_container/algorithm_mode/train.py

```diff
@@ -232,6 +234,8 @@ def train_job(train_cfg, train_dmatrix, val_dmatrix, train_val_dmatrix, model_di
 
         else:
             num_cv_round = train_cfg.pop("_num_cv_round", 1)
+            additional_output_path = os.environ[SM_OUTPUT_DATA_DIR]
```

Comment:
have you confirmed if this ENV var is always set? Might be a good idea to fallback to a sane default here. Alternatively, we'd need to check later if features are required and this is not set, then throw error there.

Reply:
I think this should be always set by the training platform `https://github.com/aws/sagemaker-containers#sm-output-data-dir`

Reply:
I will double check